src/plugins/dashboard/public/dashboard_app/top_nav/share/show_share_modal.tsx

```diff
@@ -127,40 +127,51 @@ export function ShowShareModal({
 
   let unsavedStateForLocator: DashboardLocatorParams = {};
 
-  const { dashboardState: unsavedDashboardState, panels: unsavedPanelStates } =
+  const { dashboardState: unsavedDashboardState, panels: panelModifications } =
     dashboardBackup.getState(savedObjectId) ?? {};
 
-  const latestPanels = getDashboardState().panels;
-  const unsavedPanelsMap = unsavedPanelStates
-    ? Object.entries(unsavedPanelStates).reduce((acc, [panelId, unsavedPanel]) => {
-        if (unsavedPanel && latestPanels?.[panelId]) {
-          acc[panelId] = {
-            ...latestPanels[panelId],
-            explicitInput: {
-              ...latestPanels?.[panelId].explicitInput,
-              ...unsavedPanel,
-              id: panelId,
-            },
-          };
-        }
-        return acc;
-      }, {} as DashboardPanelMap)
-    : {};
-
-  const allPanels: DashboardPanelMap = {
-    ...(unsavedDashboardState?.panels ?? {}),
-    ...unsavedPanelsMap,
-  };
+  const allUnsavedPanels = (() => {
+    if (
+      Object.keys(unsavedDashboardState?.panels ?? {}).length === 0 &&
+      Object.keys(panelModifications ?? {}).length === 0
+    ) {
+      // if this dashboard has no modifications or unsaved panels return early. No overrides needed.
+      return;
+    }
+
+    const latestPanels = getDashboardState().panels;
+    // apply modifications to panels.
+    const modifiedPanels = panelModifications
+      ? Object.entries(panelModifications).reduce((acc, [panelId, unsavedPanel]) => {
+          if (unsavedPanel && latestPanels?.[panelId]) {
+            acc[panelId] = {
+              ...latestPanels[panelId],
+              explicitInput: {
+                ...latestPanels?.[panelId].explicitInput,
+                ...unsavedPanel,
+                id: panelId,
+              },
+            };
+          }
+          return acc;
+        }, {} as DashboardPanelMap)
+      : {};
+
+    // this dashboard has unsaved panels.
+    const allUnsavedPanelsMap = {
```

Comment:
Should `allUnsavedPanelsMap` be renamed since it contains saved and unsaved panel state?

Reply:
Maybe a rename is not needed - just a comment explaining what this is and why/when its used

Reply:
done in [56049f4](https://github.com/elastic/kibana/pull/184873/commits/56049f45ffcdb5bec754e574a374a786e23a8f13)